pkg/kubernetes/controllerutil.go

```diff
@@ -151,12 +151,21 @@ func mutate(f controllerutil.MutateFn, key client.ObjectKey, obj runtime.Object)
 	return nil
 }
 
-// LookupResource lookup for resource to be created by separate resource controller
-func LookupResource(name, namespace string, resource runtime.Object, client client.Client, logger logr.Logger) (*reconcile.Result, error) {
-	err := client.Get(context.TODO(), types.NamespacedName{Namespace: namespace, Name: name}, resource)
+// LookupResourceForController lookup for resource to be created by separate resource controller
+func LookupResource(name, namespace string, resource runtime.Object, cwe eventlog.ControllerWithEvents, sendNotFoundEvent ...bool) (*reconcile.Result, error) {
```

Comment:
@rigazilla `sendNotFoundEvent` a bit confusing me, WDYT about `skipSendEvent`
Probably we can define  `cwe eventlog.ControllerWithEvents` as pointer and analyse it for the `nil` or not

Reply:
> @rigazilla `sendNotFoundEvent` a bit confusing me, WDYT about `skipSendEvent`

yeah maybe better  

> Probably we can define `cwe eventlog.ControllerWithEvents` as pointer and analyse it for the `nil` or not

this doesn't allow a 'per call' choice, only at controller level